model/parts/policies_oracles.py

```diff
@@ -0,0 +1,25 @@
+
+"""
+# Oracles related Update and Policy functions
+"""
+from model.generators.oracles import OracleRateGenerator
+from model.parts.buy_and_sell import buckets_should_be_reset
+from model.utils.generator_container import inject
+
+
+@inject(OracleRateGenerator)
+def p_oracle_report(
+    params,
+    _substep,
+    state_history,
+    prev_state,
+    oracle_rate_generator: OracleRateGenerator,
+):
+    """
+    Updates the median rate to update the Mento buckets
+    """
+    if (prev_state['timestep'] > 1) and buckets_should_be_reset(params, prev_state):
```

Comment:
I'm not sure I understand this. Why only update the oracle rate on a bucket update, and not every block?

Reply:
is it only for saving computation? I think validator rewards are mined at oracle rate every epoch block, but not sure if this is significant whether it uses an updated oracle rate or from 1 bucket update behind

Reply:
is it only for saving computation? I think validator rewards are mined at oracle rate every epoch block, but not sure if this is significant whether it uses an updated oracle rate or from 1 bucket update behind

Reply:
+1 not sure what the intent is here, the oracles should have their own report rate, right?

Reply:
I have changed it. Check out [`update()`](https://github.com/celo-org/mento2-model/blob/a77154022fc57b670a779924312b8cf725a7b8a4/model/entities/oracle_provider.py#L25)